financial-connections/src/main/java/com/stripe/android/financialconnections/features/linkstepupverification/LinkStepUpVerificationScreen.kt

```diff
@@ -0,0 +1,296 @@
+                end = 24.dp,
+                bottom = 24.dp
+            )
+    ) {
+        Spacer(modifier = Modifier.size(16.dp))
+        Title()
+        Spacer(modifier = Modifier.size(8.dp))
+        Description(payload.email)
+        Spacer(modifier = Modifier.size(24.dp))
+        ExistingEmailSection(
+            focusRequester = focusRequester,
+            otpElement = payload.otpElement,
+            enabled = confirmVerificationAsync !is Loading
+        )
+        if (confirmVerificationAsync is Fail) {
+            Spacer(modifier = Modifier.size(4.dp))
+            FormErrorText(confirmVerificationAsync.error)
+        }
+        Spacer(modifier = Modifier.size(24.dp))
+        EmailSubtext(
+            email = payload.email,
+            isLoading = resendOtpAsync is Loading,
+            onClickableTextClick = onClickableTextClick
+        )
+    }
+}
+
+@Composable
+private fun ExistingEmailSection(
+    focusRequester: FocusRequester,
+    otpElement: OTPElement,
+    enabled: Boolean
+) {
+    StripeThemeForConnections {
+        OTPElementUI(
+            focusRequester = focusRequester,
+            enabled = enabled,
+            element = otpElement
+        )
+    }
+}
+
+@Composable
+private fun EmailSubtext(
+    email: String,
+    isLoading: Boolean,
+    onClickableTextClick: (String) -> Unit
+) {
+    Row(
+        horizontalArrangement = Arrangement.spacedBy(8.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        listOf(
+            TextResource.Text(email) to 1f,
+            TextResource.Text("•") to null,
+            TextResource.StringId(R.string.stripe_link_stepup_verification_resend_code) to null
+        ).forEach { (text, weight) ->
+            AnnotatedText(
+                modifier = if (weight != null) Modifier.weight(weight, fill = false) else Modifier,
+                text = text,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+                defaultStyle = FinancialConnectionsTheme.typography.caption.copy(
+                    color = colors.textSecondary,
+                ),
+                annotationStyles = mapOf(
+                    StringAnnotation.CLICKABLE to FinancialConnectionsTheme.typography.captionEmphasized
+                        .toSpanStyle()
+                        .copy(color = if (isLoading) colors.textSecondary else colors.textBrand),
+                ),
+                onClickableTextClick = onClickableTextClick,
+            )
+        }
+        if (isLoading) {
+            CircularProgressIndicator(
+                Modifier.size(12.dp),
+                strokeWidth = 1.dp,
+                color = colors.textSecondary
+            )
+        }
+    }
+}
+
+@Composable
+private fun Description(phoneNumber: String) {
+    AnnotatedText(
+        text = TextResource.Text(
+            stringResource(
+                R.string.stripe_link_stepup_verification_desc,
+                phoneNumber
```

Comment:
Should this be the email, rather than phone number?

Reply:
woops good catch

Reply:
fixed and screenshots regenerated (sorry about the large number of files)